board/main.c

```diff
@@ -258,40 +277,51 @@ int usb_cb_control_msg(USB_Setup_TypeDef *setup, uint8_t *resp, bool hardwired)
     // **** 0xd9: set ESP power
     case 0xd9:
       if (setup->b.wValue.w == 1U) {
-        set_esp_mode(ESP_ENABLED);
+        current_board->set_esp_gps_mode(ESP_GPS_ENABLED);
       } else if (setup->b.wValue.w == 2U) {
-        set_esp_mode(ESP_BOOTMODE);
+        current_board->set_esp_gps_mode(ESP_GPS_BOOTMODE);
       } else {
-        set_esp_mode(ESP_DISABLED);
+        current_board->set_esp_gps_mode(ESP_GPS_DISABLED);
       }
       break;
     // **** 0xda: reset ESP, with optional boot mode
     case 0xda:
-      set_esp_mode(ESP_DISABLED);
+      current_board->set_esp_gps_mode(ESP_GPS_DISABLED);
       delay(1000000);
       if (setup->b.wValue.w == 1U) {
-        set_esp_mode(ESP_BOOTMODE);
+        current_board->set_esp_gps_mode(ESP_GPS_BOOTMODE);
       } else {
-        set_esp_mode(ESP_ENABLED);
+        current_board->set_esp_gps_mode(ESP_GPS_ENABLED);
       }
       delay(1000000);
-      set_esp_mode(ESP_ENABLED);
+      current_board->set_esp_gps_mode(ESP_GPS_ENABLED);
       break;
-    // **** 0xdb: set GMLAN multiplexing mode
+    // **** 0xdb: set GMLAN (white/grey) or OBD CAN (black) multiplexing mode
     case 0xdb:
-      if (setup->b.wValue.w == 1U) {
-        // GMLAN ON
-        if (setup->b.wIndex.w == 1U) {
-          can_set_gmlan(1);
-        } else if (setup->b.wIndex.w == 2U) {
-          can_set_gmlan(2);
+      if(hw_type != HW_TYPE_BLACK_PANDA){
```

Comment:
minor: I would switch the if and else so that:
```
if (hw_type == HW_TYPE_BLACK_PANDA) {
...
} else {
...
}
```

Reply:
Fixed!